comit/src/network/protocols/announce.rs

```diff
@@ -23,31 +23,38 @@ use std::{
 /// We don't implement any connection handling here but assume that another
 /// network behaviour handles connections and peer-ID to address translation for
 /// us.
+///
+/// The type parameter `C` represents the context that is used by the caller to
+/// associate submitted announce requests with the events emitted by this
+/// NetworkBehaviour.
 #[derive(NetworkBehaviour)]
-#[behaviour(out_event = "BehaviourOutEvent", poll_method = "poll")]
+#[behaviour(out_event = "BehaviourOutEvent<C>", poll_method = "poll")]
 #[allow(missing_debug_implementations)]
-pub struct Announce {
+pub struct Announce<C>
+where
+    C: fmt::Display + Send + 'static,
```

Comment:
Previously you told me that you use qualified paths out of lazyness but here you it seems you went up to the import statements to add `fmt`, why not add `fmt::Display`?

Reply:
Because I am being inconsistent with myself! :D

Using `fmt::Display` is a habit from implementing the trait because you need `fmt::Formatter` and `fmt::Result` for it, so I usually just import `std::fmt` and I am good to go.

That is not needed here so `Display` would be better yes.